### PR TITLE
PWA: First PWA tutorial

### PR DESCRIPTION
**Pull request by estelle**
god help me!

### CODE REVIEW

File: files/en-us/web/progressive_web_apps/tutorials/intro/secure_connection/index.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+You can choose a different port number. Entering `npx http-server /user/yourName/CycleTracker -p 8787` will start local server at port `8787` if available. If not, if you enter a port number that is already being used, you will get an `address already in use` or simiar error. If successfule, entering `localhost:8787` in the browser URL bar will render the index file stored as `~/user/yourName/CycleTracker/index.html`, or will display the directory contents of `~/user/yourName/CycleTracker/` if no index file is present.
+
+This non-configurable static HTTP server suffices for our basic app. For a configurable secure local web server served over HTTPs, you will need to [add a built-in SSL certificate](https://github.com/lwsjs/local-web-server/wiki/How-to-get-the-%22green-padlock%22-using-the-built-in-certificate) to be able to serve https. With a certificate, you can now install and run[local-web-server](<https://github.com/lwsjs/local-web-server/wiki/How-to-launch-a-secure-local-web-server-(HTTPS)>) from the command line to serve your project locally over https without a security warning.
```
Comment (captainbrosset) [inline files/en-us/web/progressive_web_apps/tutorials/intro/secure_connection/index.md:54]:
```suggestion
This non-configurable static HTTP server suffices for our basic app. For a configurable secure local web server served over HTTPs, you will need to [add a built-in SSL certificate](https://github.com/lwsjs/local-web-server/wiki/How-to-get-the-%22green-padlock%22-using-the-built-in-certificate) to be able to serve https. With a certificate, you can now install and run [local-web-server](https://github.com/lwsjs/local-web-server/wiki/How-to-launch-a-secure-local-web-server-(HTTPS)) from the command line to serve your project locally over https without a security warning.
```

Comment (captainbrosset) [reply files/en-us/web/progressive_web_apps/tutorials/intro/secure_connection/index.md:54]: I'm not convinced we need to talk about configuring https with local-web-server.
localhost and 127.0.0.1 are exempt from https and always considered secure. So, a reader would only ever be interested in configuring https if they're setting things up for production, with an actual domain name. But if they are, I hope they're not using this npm package. There are more robust production servers they should probably be using instead.

File: files/en-us/web/progressive_web_apps/tutorials/intro/secure_connection/index.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+## Secure external server
+
+The previous options are fine, and necessary, for testing your application as you progress thru this PWA tutorial, or any web development project. While you can host your web app on your device and make it available to anyone with an Internet connection, this is not recommended.
+
+To get the added features of PWAs, including single click installation, a standalone UI, and admission to app stores, it needs to be a PWA, which means it needs a service worker, which means we will need a secure connection. To distribute your app, enabling others to view, use, and install your PWA, you'll want to have your content hosted and available on a secure _remote_ server.
+
+When officially publishing a PWA, you will likely want to invest in a [domain name and web hosting](/en-US/docs/Learn/Common_questions/Tools_and_setup/How_much_does_it_cost#hosting). For open source projects, where developers can learn from the codebase and even contribute back to the project, you can host your progress on [GitHub Pages](https://pages.github.com/).
+
+## Github pages
```
Comment (captainbrosset) [inline files/en-us/web/progressive_web_apps/tutorials/intro/secure_connection/index.md:75]:
```suggestion
## GitHub pages
```

File: files/en-us/web/progressive_web_apps/tutorials/intro/secure_connection/index.md [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+The current state of the cycleTracker application is viewable on Github, served securely at [https://mdn.github.io/pwa-examples/cycletracker/html_and_css]. We've posted the files to the MDN Github account. Similarly, if you have a [Github](https://github.com) account, you can post it to yours. Just note that actions on Github, while securely served over SSL, are not necessarily private and all Github pages are public. If you live in an area with a repressive government that tracks menstrual cycles, like areas in Ghana and the United States, consider copying and pasting the code rather than forking it.
```
Comment (captainbrosset) [inline files/en-us/web/progressive_web_apps/tutorials/intro/secure_connection/index.md:77]:
```suggestion
The current state of the cycleTracker application is viewable on GitHub, served securely at [https://mdn.github.io/pwa-examples/cycletracker/html_and_css]. We've posted the files to the MDN GitHub account. Similarly, if you have a [Github](https://github.com) account, you can post it to yours. Just note that actions on GitHub, while securely served over SSL, are not necessarily private and all GitHub pages are public. If you live in an area with a repressive government that tracks menstrual cycles, like areas in Ghana and the United States, consider copying and pasting the code rather than forking it.
```

I suggest doing a search/replace. GitHub is the official casing for the brand, and we should probably make sure we use it consistantly.

File: files/en-us/web/progressive_web_apps/tutorials/intro/html_and_css/index.md [hunk 4]
```diff
@@ -0,0 +1,343 @@
...
+slug: Web/Progressive_web_apps/Tutorials/Intro/HTML_and_CSS
+---
+
+{{PreviousMenuNext("Web/Progressive_web_apps/Tutorials/Intro/", "Web/Progressive_web_apps/Tutorials/Intro/Secure_connection", "Web/Progressive_web_apps/Tutorials/Intro")}}
+
+{{PWASidebar}}
+
+To build a PWA, a progressive web application, we need to create a fully-functioning web application. In this section, we will markup the HTML for a static web page and enhance the appearance with CSS.
+
+Our project is to create CycleTracker, a menstrual cycle tracker.
```
Comment (captainbrosset) [inline files/en-us/web/progressive_web_apps/tutorials/intro/html_and_css/index.md:12]: I noticed that in some places, the app is written as CycleTracker, and in other places as cycleTracker (lowercase).
Can we do a global search/replace to make sure the same casing is used throughout?

File: files/en-us/web/progressive_web_apps/tutorials/intro/html_and_css/index.md [hunk 5]
```diff
@@ -0,0 +1,343 @@
...
+
+We then include an empty {{HTMLElement("section")}}. This container will be populated using JavaScript.
+
+```html
+<section id="past-periods"></section>
+```
+
+When the user submits the form, we will use JavaScript to capture the data and present a list of past periods along with a header for the section.
+
+For the time being, we temporarily hardcode some content within this `<section>`, includibg an `<h2>` header and a few past periods, to have something to style as we write the page's CSS.
```
Comment (captainbrosset) [inline files/en-us/web/progressive_web_apps/tutorials/intro/html_and_css/index.md:178]:
```suggestion
For the time being, we temporarily hardcode some content within this `<section>`, including an `<h2>` header and a few past periods, to have something to style as we write the page's CSS.
```

File: files/en-us/web/progressive_web_apps/tutorials/intro/index.md [hunk 6]
```diff
@@ -0,0 +1,93 @@
+---
+title: Creating your first PWA
+slug: Web/Progressive_web_apps/Tutorials/Intro
+---
+
+{{NextMenu("Web/Progressive_web_apps/Tutorials/Intro/HTML_and_CSS")}}
+
+{{PWASidebar}}
+
+This intro-level tutorial walks thru all the steps of building a basic progressive web app, or PWA. We will be using the web technologies we all know and love — HTML, CSS, and JavaScript — to build a period tracking web app called "CycleTracker". Like all web apps, CycleTracker is designed to work in all browsers on all devices.
```
Comment (captainbrosset) [inline files/en-us/web/progressive_web_apps/tutorials/intro/index.md:10]:
```suggestion
This intro-level tutorial walks thru all the steps of building a basic progressive web app, or PWA. We will be using web technologies — HTML, CSS, and JavaScript — to build a period tracking web app called "CycleTracker". Like all web apps, CycleTracker is designed to work in all browsers on all devices.
```

I'm hoping people who come from other app development background also use this tutorial to get into PWA development. And those people might not already know and love these languages.

File: files/en-us/web/progressive_web_apps/tutorials/intro/index.md [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+
+We will walk thru the steps of building a fully functional web app, then progressively enhance CycleTracker to make it installable and to work even when the user is offline.
+
+By default, PWAs are regular websites, built with the same technologies. Just like a regular websites, PWAs are linkable, discoverable via search engines, and visible in a browser. By including a manifest file and service worker, and serving the website over SSL, any website can become a PWA.
+
+## PWA benefits
+
+Using the languages of the web, we will create a fully functioning application that works both online and offline, both in the browsers and on the user's operating systems (OS). Like any regular website, CycleTracker can be hosted on and downloadable from a web server. All we need is a text editor: CycleTracker, like all PWAs, doesn't requie require any additional programming language knowledge, packaging, or proprietary SDK. CycleTracker, like any PWA, can be seamlessly installed on any operating system without the need of app stores (nor app store approval and fees).
```
Comment (captainbrosset) [inline files/en-us/web/progressive_web_apps/tutorials/intro/index.md:18]: > Like any regular website, CycleTracker can be hosted on and downloadable from a web server

I would replace "can be" by "is" in this case. That's how websites work, it's not really an option. And, again trying to make this as welcoming and understandable as possible to non-web people, I think it's always important to make sure they understand the distribution model.

Comment (captainbrosset) [reply files/en-us/web/progressive_web_apps/tutorials/intro/index.md:18]:
```suggestion
Using the languages of the web, we will create a fully functioning application that works both online and offline, both in the browsers and on the user's operating systems (OS). Like any regular website, CycleTracker is hosted on and downloadable from a web server. All we need is a text editor: CycleTracker, like all PWAs, doesn't require any additional programming language knowledge, packaging, or proprietary SDK. CycleTracker, like any PWA, can be seamlessly installed on any operating system without the need of app stores (nor app store approval and fees).
```

File: files/en-us/web/progressive_web_apps/tutorials/intro/index.md [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+
+- Use open web technologies you already know
```
Comment (captainbrosset) [inline files/en-us/web/progressive_web_apps/tutorials/intro/index.md:20]:
```suggestion
- Use standard and open web technologies
```

File: files/en-us/web/progressive_web_apps/tutorials/intro/index.md [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+
+  - : Historically, for an application to be installable on an OS, such as Windows, iOS, MacOS, Linux, and Android, the applications are developed in OS-supported programming languages, like C#, .Net, Objective C, Swift, Kotlin, Java, or Python. PWAs solve this problem: they use a single code base, written using regular open web technologies: HTML, CSS, and JavaScript.
```
Comment (captainbrosset) [inline files/en-us/web/progressive_web_apps/tutorials/intro/index.md:22]:
```suggestion
  - : Historically, for an application to be installable on an OS, such as Windows, iOS, MacOS, Linux, and Android, the applications are developed in OS-supported programming languages, like C#, .Net, Objective C, Swift, Kotlin, Java, or Python. PWAs are based on a different model: they use a single code base, written using standard open web technologies that work across OS: HTML, CSS, and JavaScript.
```

I would refrain from saying that using these languages is a problem to be solved. For sure, PWA is an awesome option that makes it possible to use one codebase. But it comes with its pros and cons. And going "native" is also a perfectly good options. It depends on the context.

File: files/en-us/web/progressive_web_apps/tutorials/intro/index.md [hunk 10]
```diff
@@ -0,0 +1,93 @@
...
+
+- No compiling required
+
+  - : Historically, in most cases, the application code had to be compiled and packaged into an installable format, like .exe, .dmg, .elf, and .apk, or other installable files. Depending on the language, compiling and packaging required the OS's {{glossary("SDK")}}. PWAs solve this problem. Yes, developer teams can have complex build systems, but, as we will demonstrate as we build CycleTracker, PWAs can be built out of just HTML and JavaScript (and CSS, though styling is not necessarily required for a PWA).
```
Comment (captainbrosset) [inline files/en-us/web/progressive_web_apps/tutorials/intro/index.md:26]: I would rephrase this a bit too, similarly to the previous paragraph.
Maybe the word "historically" isn't the best way to frame this. Using OS-native languages and having to compile isn't something of the past (which "historically" makes me think about). And it's not a problem either.
Again, these are just options available to developers. I think our goal is to clearly explain the ins and outs of PWAs, and their benefits, and let readers make their own choices, based on their own specific project constraints.
Our docs should help them understand that having a single codebase is, indeed, a very powerful thing, but we shouldn't have to say that the other options are in the past and problematic.

File: files/en-us/web/progressive_web_apps/tutorials/intro/index.md [hunk 11]
```diff
@@ -0,0 +1,93 @@
...
+
+- Available anywhere and everywhere
+
+  - : Historically, single-OS-only applications have bee distributed to users through downloads, often in proprietary app stores. They are available through a vendor like the Apple App Store, [Google Play](https://play.google.com/store/apps), [Microsoft Store App](https://apps.microsoft.com/store/apps), or similar. PWAs don't have these requirements. If you want to distribute your CycleTracker app, you won't require an intermediary. A user can be acquired by visiting the online version of the app.But if you want, the Play Store and App Store do distribute PWAs, undifferentiated from other iOS or Android apps.
```
Comment (captainbrosset) [inline files/en-us/web/progressive_web_apps/tutorials/intro/index.md:30]:
```suggestion
  - : Single-OS-only applications are distributed to users through downloads, often in proprietary app stores. They are available through a vendor like the Apple App Store, [Google Play](https://play.google.com/store/apps), [Microsoft Store App](https://apps.microsoft.com/store/apps), or similar. PWAs don't have these requirements. If you want to distribute your CycleTracker app, you won't require an intermediary. A user can access your app visiting its online version. But if you want, it is possible to distribute your PWA on the Play Store and App Store, undifferentiated from other iOS or Android apps.
```

File: files/en-us/web/progressive_web_apps/tutorials/intro/index.md [hunk 12]
```diff
@@ -0,0 +1,93 @@
...
+
+- Easy for the user to install
+  - : Historically, downloaded single-OS-only applications have to be intentionally installed by the user. Depending on the OS, install format, and download origin, this can be a multi-step installation process. PWAs are streamlined. PWAs are available to anyone with a supporting browser, [installable](/en-US/docs/Web/Progressive_web_apps/Guides/Installing) with a couple of clicks.
+
+### Compared to OED applications
```
Comment (captainbrosset) [inline files/en-us/web/progressive_web_apps/tutorials/intro/index.md:35]: Maybe explain what OED means.

File: files/en-us/web/progressive_web_apps/tutorials/intro/index.md [hunk 13]
```diff
@@ -0,0 +1,93 @@
...
+
+- [JavaScript and LocalStorage](/en-US/docs/Web/Progressive_web_apps/Tutorials/Intro/JavaScript_functionality)
+
+  - : Full explanation of the JavaScript functionality to make a client-side period tracker web application so we have a functioning application that can be progressively enhanced into a PWA, using `localStorage` to store period information.
+
+- [Manifest: identity, appearance, and iconography](/en-US/docs/Web/Progressive_web_apps/Tutorials/Intro/Manifest_file)
+
+  - : A PWA requires a manifest, which is a JSON file that describes the name, icon, description, and other definitions of how the application works within the operating system on which the PWA is installed. We will create a manifest file that defines the appearance of the application when installed, including which icons should be used depending on the user's device, and parameters for the application viewport. We also look at debuggin the manifest file with browser developer tools.
```
Comment (captainbrosset) [inline files/en-us/web/progressive_web_apps/tutorials/intro/index.md:71]:
```suggestion
  - : A PWA requires a manifest, which is a JSON file that describes the name, icon, description, and other definitions of how the application works within the operating system on which the PWA is installed. We will create a manifest file that defines the appearance of the application when installed, including which icons should be used depending on the user's device, and parameters for the application viewport. We also look at debugging the manifest file with browser developer tools.
```

**Review by captainbrosset — CHANGES_REQUESTED** [review]
Wow, that's a lot of work. Thank you for being so thorough.
Apologies for the number of comments I made. I truly hope they're useful. Let me know if you have any questions.

File: files/en-us/web/progressive_web_apps/tutorials/intro/service_workers/index.md [hunk 14]
```diff
@@ -0,0 +1,395 @@
...
+
+![Two service workers exist at localhost:8080. The can be be unregistered from the list of service workers](edge_sw_list.jpg)
+
+The service worker panel provides access to a list of all the registered service workers for the browser; not just the service worker for PWA opened in the current tab. This list is another way to stop, start, or unregister any service worker.
```
[the remark this answers is not in the document]
Comment (estelle) [reply files/en-us/web/progressive_web_apps/tutorials/intro/service_workers/index.md:385]: The link is inside what I call "the devtools applications panel". This is what i was referring to. Is there a better term? Maybe this is clearer:

```suggestion
The service worker window within the DevTools' application panel, provides a link to access to pop up window containing a list of all the registered service workers for the browser; not just the service worker for the application opened in the current tab. Each service worker list of workers has buttons to stop, start, or unregister that individual service worker.
```

File: files/en-us/web/progressive_web_apps/tutorials/intro/service_workers/index.md [hunk 15]
```diff
@@ -0,0 +1,395 @@
...
+        return cachedResponse;
+      } else {
+        // If resource isn't in the cache, return a 404.
+        return new Response(null, { status: 404 });
+      }
+    })()
+  );
+});
+```
+
```
Comment (estelle) [inline files/en-us/web/progressive_web_apps/tutorials/intro/service_workers/index.md:298]:
```suggestion
When updating a service worker, the VERSION constant doesn't need to be updated, as any change in the content of the service worker script itself will trigger the browser to install the new service worker. However, it is a good practice to update the version number as it makes it easier for devs, including yourself, to see which version of the service worker is currently running in the browser, by [checking the name of the Cache in the Application tool](#with_developer_tools) (or Sources tool).

**Note:** Updating VERSION is important when making changes to any application resource, including the CSS, HTML, and JS code, and image assets. The version number, or any change to the service worker file, is the only way to force an update of the app for your users.
```

File: files/en-us/web/progressive_web_apps/tutorials/intro/index.md (hunk 10 above)
Comment (estelle) [reply files/en-us/web/progressive_web_apps/tutorials/intro/index.md:26]: "With most programming languages, like Java, C/C++, and Kotlin, which are commonly used for building Android apps, and Objective-C and Swift, for iOS, the code needs to be compiled and packaged into an installable format, like .exe, .dmg, .elf, and .apk, or another installable file type, depending on the operating system. Depending on the language, compiling and packaging may require the OS's {{glossary("SDK")}}. PWAs use web technologies supported by every operating system that doesn't need to be packaged or compiled. Yes, developer teams can have complex build systems, but, as we will demonstrate as we build CycleTracker, PWAs can be built out of just HTML and JavaScript (and CSS, though styling is not necessarily required for a PWA)."

File: files/en-us/web/progressive_web_apps/tutorials/intro/index.md (hunk 12 above)
Comment (estelle) [reply files/en-us/web/progressive_web_apps/tutorials/intro/index.md:35]: used Compared to OS-installed software applications insted

File: files/en-us/web/progressive_web_apps/tutorials/intro/service_workers/index.md [hunk 16]
```diff
@@ -0,0 +1,395 @@
...
+slug: Web/Progressive_web_apps/Tutorials/Intro/Service_workers
+---
+
+{{PWASidebar}}
+
+{{PreviousMenu("Web/Progressive_web_apps/Tutorials/Intro/Manifest_file", "Web/Progressive_web_apps/Tutorials/Intro")}}
+
+Thus far, we've written the HTML, CSS, and JavaScript for cycleTracker. We added a manifest file defining colors, icons, URL, and other app features. We have a working web app! But it isn't yet a PWA. In this section, we will write the JavaScript required to convert our fully functional web application into a PWA that can be distributed as a standalone app and works seamlessly offline.
+
+If you haven't already done so, copy the [HTML](https://github.com/mdn/pwa-examples/tree/master/cycletracker/manifest_file/index.html), [CSS](https://github.com/mdn/pwa-examples/tree/master/cycletracker/manifest_file/style.css), and [manifest](https://github.com/mdn/pwa-examples/tree/master/cycletracker/manifest_file/cycletracker.json) JSON file. Save them to files called `index.html`, `styles.css`, and `cycletracker.json`.
```
Comment (estelle) [inline files/en-us/web/progressive_web_apps/tutorials/intro/service_workers/index.md:12]:
```suggestion
If you haven't already done so, copy the [HTML](https://github.com/mdn/pwa-examples/tree/master/cycletracker/manifest_file/index.html), [CSS](https://github.com/mdn/pwa-examples/tree/master/cycletracker/manifest_file/style.css), [JavaScript](https://github.com/mdn/pwa-examples/tree/master/cycletracker/manifest_file/app.js), and [manifest](https://github.com/mdn/pwa-examples/tree/master/cycletracker/manifest_file/cycletracker.json) JSON file. Save them to files called `index.html`, `styles.css`, `app.js`, and `cycletracker.json`, respectively.
```

File: files/en-us/web/progressive_web_apps/tutorials/intro/service_workers/index.md [hunk 17]
```diff
@@ -0,0 +1,395 @@
...
+
+We achieve these tasks by reacting to three service worker events, including the
+
+- [`fetch`](/en-US/docs/Web/API/ServiceWorkerGlobalScope/fetch_event),
+- [`install`](/en-US/docs/Web/API/ServiceWorkerGlobalScope/install_event), and
+- [`activate`](/en-US/docs/Web/API/ServiceWorkerGlobalScope/activate_event) events.
+
+### Version number
+
+Once the PWA is installed on the user's machine, the only way to guarantee updated files are retrieved is for there to be a change in the service worker. If a change is made to any other PWA resource — if the HTML is updated, a bug is fixed in the CSS, a function is added to `app.js`, an image is compressed to reduce the file save, etc. — the installed PWA will not know it needs to download updated resources. When the service worker is altered in any way, the PWA will know it should update the cache.
```
Comment (estelle) [inline files/en-us/web/progressive_web_apps/tutorials/intro/service_workers/index.md:40]:
```suggestion
Once the PWA is installed on the user's machine, the only way to inform the browser that there are updated files to be retrieved is for there to be a change in the service worker. If a change is made to any other PWA resource — if the HTML is updated, a bug is fixed in the CSS, a function is added to `app.js`, an image is compressed to reduce the file save, etc. — the service worker of your installed PWA will not know it needs to download updated resources. Only when the service worker is altered in any way, will the PWA know it may be time to update the cache; which is the service worker's task to initiate.
```

File: files/en-us/web/progressive_web_apps/tutorials/intro/service_workers/index.md [hunk 18]
```diff
@@ -0,0 +1,395 @@
...
+We get the names of the existing named caches. We use the {{domxref("CacheStorage.keys()")}} method (again accessing `CacheStorage` through the global {{domxref("caches")}} property) which returns a {{jsxref("Promise")}} that resolves with an array containing strings corresponding to all of the named {{domxref("Cache")}} objects in the order they were created.
+
+We use the [`Promise.all()`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all) method is to iterate thru that list name promises. The `all()` method takes as input a list of iterable promises and returns a single Promise. For each name in the list of named cachees, check if the cache is the currently active cache. If not, delete it with the `Cache` [`delete()`](/en-US/docs/Web/API/Cache/delete) method.
+
+The last line, the `await clients.claim()` uses the [`claim()`](/en-US/docs/Web/API/Clients/claim) method of the [`Clients`](/en-US/docs/Web/API/Clients) interface to enable our service worker to set itself as the controller for our PWA. The `claim()` method enables the service worker to "claim control" of all clients within its scope. This way, clients loaded in the same scope don't need to be reloaded.
```
Comment (estelle) [inline files/en-us/web/progressive_web_apps/tutorials/intro/service_workers/index.md:168]:
```suggestion
The last line, the `await clients.claim()` uses the [`claim()`](/en-US/docs/Web/API/Clients/claim) method of the [`Clients`](/en-US/docs/Web/API/Clients) interface to enable our service worker to set itself as the controller for our client; the "client" referring to a running instance of the PWA. The `claim()` method enables the service worker to "claim control" of all clients within its scope. This way, clients loaded in the same scope don't need to be reloaded.
```

**Comment by estelle** [comment]
Thanks @captainbrosset for all the feedback. I think I addressed everything.

File: files/en-us/web/progressive_web_apps/tutorials/intro/secure_connection/index.md [hunk 19]
```diff
@@ -0,0 +1,93 @@
...
+
+With the `index.html` updated, and the `style.css` housed in the same directory, viewing the page in a narrow browser window should look similar to this screenshot:
+
+![Light green web page with a large header, a form with a legend, two date pickers and a button. The bottom shows two placeholder menstrual cycles and a header.](filefile.jpg)
+
+We are viewing our page using the `file://` protocol. This works for the current state of our codebase, and will continue to suffice as we [add JavaScript functionality](/en-US/docs/Web/Progressive_web_apps/Tutorials/Intro/JavaScript_functionality). However, manifest files and services workers, both PWA requirements, require a secure connection. PWAs need to served from a web server over `https` or a local development environment, using `localhost`, `127.0.0.1`, with or without a port number. If we view our finished app using the `files://` protocol, our [manifest file](/en-US/docs/Web/Progressive_web_apps/Tutorials/Intro/Manifest_file) will be ignored and any [service workers](/en-US/docs/Web/Progressive_web_apps/Tutorials/Intro/Service_workers) we add will fail.
```
[the remark this answers is not in the document]
Comment (estelle) [reply files/en-us/web/progressive_web_apps/tutorials/intro/secure_connection/index.md]: ```> **Note:** Serving your app over `https` isn't only good for PWAs, but for all websites as it ensures the information that transits between your web server and the user's browser is encrypted end to end. Several [Web APIs require secure contexts](/en-US/docs/Web/Security/Secure_Contexts/features_restricted_to_secure_contexts). Even if you aren't creating installable PWAs, as you add features to any web app, you may run into cases where a secure context is required.``

**Comment by captainbrosset** [comment]
Sorry about the delay. I will give this another look today.

**Review by captainbrosset — APPROVED** [review]
Here are my final comments.

Please take a look. A lot of them are minor suggestions, and a couple are a bit more important.

On top of these comments, I wanted to say that the links to the CycleTracker app on github.io don't all seem to work for me. It's perhaps expected since I know there's a PR waiting on the samples repo. But I just wanted to make sure you knew.

Let me end by saying thank you for the very thorough work. I'll approve now because I don't need to re-review again after you've made some final changes based on my comments, or decided to ignore them (which you can safely do for some of them).

File: files/en-us/web/progressive_web_apps/tutorials/intro/index.md [hunk 20]
```diff
@@ -0,0 +1,93 @@
...
+
+Once installed, PWAs can be made to appear and act similarly to other applications installed on the user's device:
+
+- Application window
+
+  - : With a [manifest](/en-US/docs/Web/Progressive_web_apps/Tutorials/Intro/Manifest_file#app_presentation) setting, we'll make CycleTracker open in its own application window, meaning installed PWAs are similar to other installed applications.
+
+- Application icon
+
+  - : PWAs display an application icon in the same location as other installed applications installed on the users' operating system. This can be an icon on the homescreen, in the toolbar, in the application's folder, or wherever the device displays application icons. We'll learn how to [declare icons](/en-US/docs/Web/Progressive_web_apps/Tutorials/Intro/Manifest_file#app_iconography) for CycleTracker, so, once installed, our PWA can appear like any other OED application on the users device.
```
Comment (estelle) [inline files/en-us/web/progressive_web_apps/tutorials/intro/index.md:45]:
```suggestion
  - : PWAs display an application icon in the same location as other installed applications installed on the users' operating system. This can be an icon on the homescreen, in the toolbar, in the application's folder, or wherever the device displays application icons. We'll learn how to [declare icons](/en-US/docs/Web/Progressive_web_apps/Tutorials/Intro/Manifest_file#app_iconography) for CycleTracker, so, once installed, our PWA can appear like any other installed application on the user's device.
```

File: files/en-us/web/progressive_web_apps/tutorials/intro/javascript_functionality/index.md [hunk 21]
```diff
@@ -0,0 +1,320 @@
+---
+title: JavaScript web functionality
```
Comment (estelle) [inline files/en-us/web/progressive_web_apps/tutorials/intro/javascript_functionality/index.md:2]:
```suggestion
title: JavaScript functionality
```

File: files/en-us/web/progressive_web_apps/tutorials/intro/javascript_functionality/index.md [hunk 22]
```diff
@@ -0,0 +1,320 @@
...
+slug: Web/Progressive_web_apps/Tutorials/Intro/JavaScript_functionality
+---
+
+{{PWASidebar}}
+
+{{PreviousMenuNext("Web/Progressive_web_apps/Tutorials/Intro/HTML_and_CSS", "Web/Progressive_web_apps/Tutorials/Intro/Manifest_file", "Web/Progressive_web_apps/Tutorials/Intro")}}
```
Comment (estelle) [inline files/en-us/web/progressive_web_apps/tutorials/intro/javascript_functionality/index.md:8]:
```suggestion
{{PreviousMenuNext("Web/Progressive_web_apps/Tutorials/Intro/Secure_connection", "Web/Progressive_web_apps/Tutorials/Intro/Manifest_file", "Web/Progressive_web_apps/Tutorials/Intro")}}
```

File: files/en-us/web/progressive_web_apps/tutorials/intro/secure_connection/index.md [hunk 23]
```diff
@@ -0,0 +1,95 @@
...
+
+When officially publishing a PWA, you will likely want to invest in a [domain name and web hosting](/en-US/docs/Learn/Common_questions/Tools_and_setup/How_much_does_it_cost#hosting). For open source projects, where developers can learn from the codebase and even contribute back to the project, you can host your progress on [GitHub Pages](https://pages.github.com/).
+
+## GitHub pages
+
+The current state of the CycleTracker application is viewable on GitHub, served securely at [https://mdn.github.io/pwa-examples/cycletracker/html_and_css]. We've posted the files to the MDN GitHub account. Similarly, if you have a [GitHub](https://github.com) account, you can post it to yours. Just note that actions on GitHub, while securely served over SSL, are not necessarily private and all GitHub pages are public. If you live in an area with a repressive government that tracks menstrual cycles, like areas in Ghana and the United States, consider copying and pasting the code rather than forking it.
```
Comment (estelle) [inline files/en-us/web/progressive_web_apps/tutorials/intro/secure_connection/index.md:79]:
```suggestion
The current state of the CycleTracker application is viewable on GitHub, served securely at [https://mdn.github.io/pwa-examples/cycletracker/html_and_css](https://mdn.github.io/pwa-examples/cycletracker/html_and_css). We've posted the files to the MDN GitHub account. Similarly, if you have a [GitHub](https://github.com) account, you can post it to yours. Just note that while securely served over SSL, actions on GitHub are not necessarily private, and all GitHub pages are public. If you live in an area with a repressive government that tracks menstrual cycles, consider copying and pasting the code rather than forking it.
```
